addon/services/m3-schema-manager.js

```diff
@@ -21,7 +23,16 @@ export default class SchemaManager extends Service {
    * @returns {Object}
    */
   computeAttributeReference(key, value, modelName, schemaInterface) {
-    return this.get('schema').computeAttributeReference(key, value, modelName, schemaInterface);
+    let schema = this.get('schema');
+    if (this.useComputeAttribute()) {
+      let computedValue = schema.computeAttribute(key, value, modelName, schemaInterface);
+      let valueType = schemaTypesInfo.get(computedValue);
+      if (valueType === REFERENCE) {
+        return computedValue;
+      }
+    } else {
+      return schema.computeAttributeReference(key, value, modelName, schemaInterface);
+    }
```

Comment:
I would not expect someone opted in to the new API to end up with call paths going through the old API.  Is there a good reason for this?

Reply:
Good point, fixed